backend/src/utils/validation/checkDuplicates.ts

```diff
@@ -1,14 +1,17 @@
 import { IClientCourse } from 'interfaces/Ientities/IclientCourses';
+import CourseStatus from 'enums/coursesEnums';
 
-const checkCourseDuplicates = (courseArr: IClientCourse[], courseId: string) => {
-  const arr = courseArr.find((clientCourse) => {
-    if ('_id' in clientCourse.course && clientCourse.course._id !== undefined) {
-      return clientCourse.course._id.toString() === courseId;
+const checkCourseDuplicates = (courseArr: IClientCourse[], courseId: string): boolean => {
+  const alreadyAppliedCourse = courseArr.find((clientCourse) => {
+    if (clientCourse && '_id' in clientCourse.course && clientCourse.course._id) {
+      return (
+        clientCourse.course._id.toString() === courseId &&
+        clientCourse.status !== CourseStatus.rejected
+      );
     }
-
-    return clientCourse.course.toString() === courseId;
+    return null;

```

Comment:
Not really good idea to return `null` when you describe the result of the function as `boolean`. And well, in general, it is a really good practice to have only one `return` statement in a function, but not **3**